File: R/grid_state_correspondence_table.R
#' correspondance table indicating what quadrants of IBGE's statistical grid intersect with each Brazilian state
#'
#' @title A correspondance table indicating what quadrants of IBGE's statistical grid intersect with each Brazilian state
#' @description Built-in dataset
#'
#'
#' \itemize{
#'   \item \code{code_uf}: IBGE code of State (2-digit, numeric)
#'   \item \code{name_state}: Title-case name of state (character)
#'   \item \code{code_grid}: Unique code of each quadrant of IBGE's statistical grid
#' }
#'
#' @docType data
#' @keywords datasets
#' @name grid_state_correspondence_table
#'
#' @usage data(grid_state_correspondence_table)
#' @note Last updated 2019-06-17
#' @format A data frame sf with 139 rows and 3 columns
"grid_state_correspondence_table"

File: python-package/geobr/grid_state_correspondence_table.py
"""Correspondence table indicating what quadrants of IBGE's statistical grid intersect with each Brazilian state"""

from pathlib import Path

import pandas as pd

def _grid_state_correspondence_path() -> Path:
    return Path(__file__).parent / "data" / "grid_state_correspondence_table.csv"

def grid_state_correspondence_table() -> pd.DataFrame:
    """
    A correspondence table indicating what quadrants of IBGE's statistical grid intersect with each Brazilian state
    """

    return pd.read_csv(_grid_state_correspondence_path(), encoding='latin1')
